plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Data.hs

```diff
@@ -0,0 +1,206 @@
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Ide.Plugin.Cabal.Completion.Data where
+
+import           Data.Map                                       (Map)
+import qualified Data.Map                                       as Map
+import qualified Data.Text                                      as T
+import           Distribution.CabalSpecVersion                  (CabalSpecVersion (CabalSpecV2_2),
+                                                                 showCabalSpecVersion)
+import           Ide.Plugin.Cabal.Completion.Types
+import           Ide.Plugin.Cabal.LicenseSuggest                (licenseNames)
+
+import           Ide.Plugin.Cabal.Completion.Completer.FilePath
+import           Ide.Plugin.Cabal.Completion.Completer.Module
+import           Ide.Plugin.Cabal.Completion.Completer.Simple
+import           Ide.Plugin.Cabal.Completion.Completer.Types    (Completer)
+
+-- ----------------------------------------------------------------
+-- Completion Data
+-- ----------------------------------------------------------------
+
+-- | Keyword for cabal version; required to be the top line in a cabal file
+cabalVersionKeyword :: Map KeyWordName Completer
+cabalVersionKeyword =
+  Map.singleton "cabal-version:" $
+    constantCompleter $
+      map (T.pack . showCabalSpecVersion) [CabalSpecV2_2 .. maxBound]
```

Comment:
Document in the function, that we decided the minBound arbitrarily, there is no reason, except that we think no one should use anything older.